google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting;
+import com.google.cloud.tools.intellij.appengine.server.run.AppEngineServerConfigurationType;
+import com.google.cloud.tools.intellij.ui.BooleanTableModel;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.intellij.CommonBundle;
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.executors.DefaultRunExecutor;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.execution.util.EnvironmentVariable;
+import com.intellij.javaee.run.configuration.RunnerSpecificLocalConfigurationBit;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.BooleanTableCellEditor;
+import com.intellij.ui.BooleanTableCellRenderer;
+import com.intellij.ui.TableUtil;
+import com.intellij.ui.table.JBTable;
+import com.intellij.util.ui.UIUtil;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.IntStream;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.ListSelectionModel;
+import javax.swing.border.Border;
+import javax.swing.border.EmptyBorder;
+import javax.swing.table.DefaultTableCellRenderer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Dialog visible only in Ultimate Edition that (1) confirms the download of the service account
+ * JSON key for the Google Cloud Libraries , (2) provides information on how to set the environment
+ * variables for local run and (3) allows the user to select App Engine Standard run configurations
+ * to automatically update with these environment variables.
+ */
+public class ServiceAccountKeyUltimateDisplayDialog extends DialogWrapper {
+
+  private static final Logger LOG =
+      Logger.getInstance(ServiceAccountKeyUltimateDisplayDialog.class);
+
+  private final Project project;
+  private final String gcpProjectId;
+  private final String downloadPath;
+  private JPanel mainPanel;
+  private ServiceAccountKeyDownloadedPanel commonPanel;
+  private JTable table;
+  private JLabel label;
+  private JScrollPane scrollPane;
+  private BooleanTableModel tableModel;
```

Comment:
Possibly should be `BooleanTableModel<RunnerAndConfigurationSettings>`? It looks like it accepts type of its values?